react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+class Stopwatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isTicking: false,
+      seconds: 0
+    };
+    this.handleClickIcon = this.handleClickIcon.bind(this);
+    this.handleClickClock = this.handleClickClock.bind(this);
+  }
+
+  handleClickIcon() {
+    if (this.state.isTicking) {
+      clearInterval(this.timerID);
+    } else {
+      this.timerID = setInterval(() => this.tick(), 1000);
+    }
+    this.setState({ isTicking: !this.state.isTicking });
+  }
+
+  handleClickClock() {
+    if (!this.state.isTicking) {
+      clearInterval(this.timerID);
+      this.setState({ seconds: 0 });
+    }
+  }
+
+  tick() {
+    this.setState({ seconds: this.state.seconds + 1 });
+  }
+
+  render() {
+    const icon = this.state.isTicking ? 'pause' : 'play';
+    const clickableClock = !this.state.isTicking && this.state.seconds > 0 ? 'click' : 'noClick';
```

Comment:
Minor quibble, but these things add up:

```js
const clickableClock = !this.state.isTicking && this.state.seconds > 0 ? 'click' : 'noClick';
```

using `!` like this is really hard in compound logical expressions. Try rewriting it to avoid that.

```js
const clockClass = this.state.isTicking || this.state.seconds === 0
  ? ''
  : 'click';
```

Reply:
really hard to understand*

Reply:
Ahh yes, that looks much cleaner. Thank you!